docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,383 @@
+# Semantic Conventions for HTTP-relevant metrics emitted by .NET and ASP.NET Core
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for HTTP metrics emitted by .NET components and runtime.
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [HTTP client](#http-client)
+  * [Metric: `http.client.open_connections`](#metric-httpclientopen_connections)
+  * [Metric: `http.client.connection.duration`](#metric-httpclientconnectionduration)
+  * [Metric: `http.client.duration`](#metric-httpclientduration)
+  * [Metric: `http.client.request.time_in_queue`](#metric-httpclientrequesttime_in_queue)
+  * [Metric: `http.client.active_requests`](#metric-httpclientactive_requests)
+- [HTTP server](#http-server)
+  * [Metric: `http.server.duration`](#metric-httpserverduration)
+  * [Metric: `http.server.active_requests`](#metric-httpserveractive_requests)
+
+<!-- tocstop -->
+
+## HTTP client
+
+All Http client metrics are reported by `System.Net.Http` meter.
+
+### Metric: `http.client.open_connections`
```

Comment:
no stong opionion here. moving non-standard metrics to custom namespaces makes sense while they are non-standard, but in the long term, most of the end-users don't care which handler metric comes from they'd likely expect all handlers to produce standard metrics plus, potentially, some custom ones.

Reply:
I think of it mostly as a question of standardization. Anything named "http.client" we are saying we expect it would be standardized across implementations from different language stacks. Anything with a technology specific name (such as kestrel) we are saying its specific to .NET. I'm not sure we've been consistent in our bucketing, likely because kestrel has a distinctive technology-specific name whereas the "HttpClient" .NET tech name is conspicuously close to OTel's "http.client" language neutral name. In theory we could name things "dotnet.http.client.*" as .NET tech name but if @lmolkova thinks the language-neutral names are reasonable and likely to be standardized that way I'm happy to just implement that now rather than implementing "dotnet.http.client.*" first and then get requested to change the names to "http.client.*" later.

@lmolkova, are there any metrics of the form kestrel.blah where you expect OTel is likely to standardize http.server.blah? For example http.server.connection.duration or http.server.active_connections? Similar to HttpClient, those might be spots where we just want to  take a guess where it is going rather than start with a tech-specific name and then be requested to do a breaking change later to standardize.

Also FYI, I'm trying to keep the HttpClient part stable so that Anton can implement, get it reviewed, test, etc without us moving the goal posts on him :) For aspnet, I trust if you are proposing a change James that you are comfortable with how it would impact the work to finish up.

Reply:
OTel discourages adding custom things into 'otel' namespaces to avoid future conflicts, but if we believe that in the long term, the metric should not be .net-specific, adding it to the `dotnet` namespace does not seem right,

As @antonfirsov mentioned somewhere here, if we do `custom_prefix` and then the standard one comes up, we'd definitely have to change. If we don't do `custom_prefix` and then a standard one comes up, we'd probably need to change too because of some other mismatch. 

We can try bringing HTTP connection durations metrics to otel (after stability because of the feature freeze) and then let's keep `http.client.`. I can help with this.